index.rst

```diff
@@ -44,14 +44,579 @@
+#. The OCPS (really the butler) requires access to EFD. This is not currently captured in a use-case but one can envision how having a pipeTask be capable to correlate image quality against items in the EFD could be useful.
+
+   - No code has been written to integrate butler directly with EFD, but it is possible to do
+   - Would enable useres to define pipelines that explicitly specified EFD datasets as pipeline inputs. 
+     Currently, it would be required to sort out the mapping of Exposure dataId to and EFD call in (potentially) a special runQuantum method in the pipeline task
+  
+
+Entirely New Functionality
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section identifies functionalities that are required and could not be assertained by upgrading already existing components.
+The largest piece of missing functionalty is the framework to perform on-the-fly analyses which are triggered on specific events or conditions, then able to perform calculations, generate a report (including plots etc), and have the operator be alerted.
+Implementing this type of capability requires numerous pieces to work together.
+
+
+Catcher CSC
+'''''''''''''
+
+A series of new functionality, which for the purposes of this document we have grouped into a single "Catcher" Controllable SAL Component (CSC) is required to handle the low-level coordiation of identifying when a specific condition is met, then launching and monitoring an analysis process.
+It is still being evaluated if it is required to generate a new CSC or if the Watcher CSC can be augmented to handle this new functionality. 
+For the purposes of this document we will assume the functionality is to be captured in a new CSC.
+The Catcher functionality also requires a LOVE display to show which tasks are running, links to generated reports, and alarms or notifications for observers.
+
+More details on the design and implementation can be found in the `Catcher Design Document <https://docs.google.com/document/d/1mbmfqjebOuHIV8CwC7jFHcFKCRMtyBDHPXeGfBO1EPE/edit#>`_ currently being worked on as a google doc.
+
+At the end of an analysis task, a "report" is generated and produced as a result.
+The reports from the Catcher can be derived in multiple ways and take on multiple formats.
+There is no requirement that the analysis aspects generated by the Catcher managed tasks be persistent but it is recommended.
+When possible, this committee recommends that analysis tasks produce reports in the form of `Papermill Executed Parameterized Notebooks`_.
+Once executed, the notebooks and their contents and archived to the LFA, where they can be looked at (and even re-run) either immediately or at a later date.
+The capabilities of the notebooks is vast; allowing image analysis via sending commands to the OCPS, queries of the EFD, grabbing of SAL events or data from the Butler.
+The data to create any plots or other displays are also contained in the notebook allowing plots to be modified and re-generated as required.
```

Comment:
I did not realize that data could be associated with a notebook as oppose to the notebook loading data that exists elsewhere. If this is indeed the case, it seems potentially a powerful tool to preserve the relevant data from when a particular event/analysis took place.

Reply:
So this sentence isn't what I wanted it to say, so I fixed it.

You're correct that the notebook COULD store data, but what is more important is that it stores the info to query the data (from the EFD or whatever). Ideally, one should not have to store the data in the notebook itself.
It turns out that we're concerned about papermill because the python API documentation is really weak, so we might have to store data along with the reports via another mechanism... but I hope not.